### DISCUSSION TITLE
Disable internal buffer?

### DISCUSSION THREAD

**Discussion by nikolaiaas** [HaishinKit/HaishinKit.swift#1199] [Q&A]
Hi. Thanks for all your hard work on the project!
I have an issue with live streaming. If e.g. I minimise my app for 10 secs and open it up again, the stream will keep sending. All good!
But now all the frames that were recorded when the app was minimised will be in a buffer. So the delay from my phone to the viewer will be longer.

I larix I have unsetBuffer that I can set to the max ms of unsent video should be kept.
Is there any option for this as well with HaishinKit?

By the way, just played with HEVC. Awesome feature drop!

**Comment by shogo4405** [comment]
I don't have such an option available, particularly.
I would like to confirm if "minimise my app" refers to minimizing the window on macOS?

**Reply by nikolaiaas** [reply]
Hi
This is on ios. So e.g if the phone rings, I check notifycations or something else while streaming.
Then the video is buffered, and sent once I open the app again. For live shopping, it would be better just to skip the frames.

It also happends if there is bad connection, and the frames can't be sent fast enought.
So in the beginning of streaming there might be 5 secs delay from screen to screen. But later in the stream there could be 30 secs delay

**Reply by nikolaiaas** [reply]
This feature is the only thing I have outstanding in my project. I will happily sponser more for this feature

**Reply by shogo4405** [reply]
In the case of iOS, if such a situation occurs, the camera capture stops, so I don't know why the video buffer is getting full.

About bad connection, it can be avoided by implementing bitrate control as follows.
https://github.com/shogo4405/HaishinKit.swift/blob/main/Examples/iOS/LiveViewController.swift#L306-L309

It's also possible to drop video frames using the delegate that was recently implemented. Please try it on the main branch.
https://github.com/shogo4405/HaishinKit.swift/blob/main/Sources/Net/NetStream.swift#LL24C1-L25C58
